modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java

```diff
@@ -85,8 +86,21 @@ public int doStartTag() throws JspException {
 
 				hasChildCPDefinitions = _cpContentHelper.hasChildCPDefinitions(
 					cpDefinitionId);
+
 				_productSettingsModel = _productHelper.getProductSettingsModel(
 					cpDefinitionId);
+
+				int multipleQuantity =
+					_productSettingsModel.getMultipleQuantity();
+
+				int[] allowedQuantities = ArrayUtil.filter(
+					_productSettingsModel.getAllowedQuantities(),
+					quantity ->
+						(quantity >= _productSettingsModel.getMinQuantity()) &&
+						(quantity <= _productSettingsModel.getMaxQuantity()) &&
+						((quantity % multipleQuantity) == 0));
```

Comment:
Can you double check sourceFormat and update accordingly? You should not have to declare multipleQuantity. Confirm for me. Everything else looks good.

Reply:
I double checked, it says "Create a new variable for the left hand side operand of the '==' operator for better readability"